bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -328,6 +328,11 @@ pub mod pallet {
 			let update_has_next_sync_committee = !<NextSyncCommittee<T>>::exists() &&
 				(update.next_sync_committee_update.is_some() &&
 					update_attested_period == store_period);
+
+			println!("update_has_next_sync_committee {}", update_has_next_sync_committee);
+			println!("update.attested_header.slot {}", update.attested_header.slot);
+			println!("latest_finalized_state.slot {}", latest_finalized_state.slot);
+
 			ensure!(
 				update.attested_header.slot > latest_finalized_state.slot ||
```

Comment:
Perhaps this should rather be:
```suggestion
				update.finalized_header.slot > latest_finalized_state.slot ||
```

Reply:
It would deviate from the spec though: https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#validate_light_client_update

![Screenshot 2024-09-02 at 09 28 34](https://github.com/user-attachments/assets/3a668d10-15aa-41ac-9454-b4585b44ccd4)

Reply:
I'd prefer to not make this change deviate from the spec. To avoid free consensus update maybe we can check the update interval even for sync committee updates?

https://github.com/Snowfork/polkadot-sdk/blob/dfee885a3bdb993a2c122e5dd4549ed29183b07e/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L659

Reply:
What happens if the sync committee update slot is smaller than the current finalized header? Then using the interval check does not work

Reply:
> if an update makes it through verify_update, it is relevant and will change the light client state.

It's a normal case `update.finalized_header.slot < latest_finalized_state.slot` for sync committee updates.

Reply:
> What happens if the sync committee update slot is smaller than the current finalized header? Then using the interval check does not work

What I mean is something like this:

```
if update.next_sync_committee_update.is_some() &&
				update.finalized_header.slot >=
					latest_slot.saturating_add((T::FreeHeadersInterval::get() / 2) as u64)
			{
				return Pays::No;
			}
```

Reply:
> > if an update makes it through verify_update, it is relevant and will change the light client state.
> 
> I'd assume it's a normal case `update.finalized_header.slot < latest_finalized_state.slot` for sync committee updates. Still prefer to strictly follow the spec unless absolute nessessary to change.

Yes, but if `update.finalized_header.slot < latest_finalized_state.slot` then the update must contain a sync committee update and the NextSyncCommittee<T> must be empty: https://github.com/paritytech/polkadot-sdk/blob/master/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L332

For the case I highlighted in the PR description, `update_has_next_sync_committee` is false and the finalized update is older than the finalized header.

I understand your reluctance to deviate from the spec, but fundamentally it is worrying to be that an update is essentially irrelevant but can somehow pass the validation and make it to the apply function.

Reply:
> > What happens if the sync committee update slot is smaller than the current finalized header? Then using the interval check does not work
> 
> What I mean is something like this to avoid spamming.
> 
> ```
> 			if update.next_sync_committee_update.is_some() &&
> 				update.finalized_header.slot >=
> 					latest_slot.saturating_add((T::FreeHeadersInterval::get() / 2) as u64)
> 			{
> 				return Pays::No;
> 			}
> ```

I don't understand the logic? Why divide `FreeHeadersInterval` by 2 here?

Reply:
To add some loose check for sync committee update avoid spamming, just an example. 

> // If the next sync committee is not known and this update sets it, the update is free.
> // If the sync committee update is in a period that we have not received an update for,
> // the update is free.

I like the impl in your recent commit.